### PR TITLE
Update rules_pkg to preserve old behavior

### PR DESCRIPTION
**Pull request by GordonHuangKodiak**
### Description

Reimplement Kodiak custom arguments, modify tarball paths to have absolute and local path similar to previous version.

### Associated Jira tickets

https://jira.internal.kodiak.ai/browse/ENG-7907

### Design document

N/A

### Status

- [ ] Unit test implemented.
- [ ] Includes/impacts simulation scenario.
- [ ] Includes viewer related changes.
- [ ] Simulation assessment performed.
- [x ] Dev test performed.
- [ ] Should impact MPI.

**Operational Impact:**

Tarball or DEB file install may fail if path is slightly different.

### CODE REVIEW

File: pkg/private/pkg_files.bzl [hunk 1]
```diff
@@ -515,7 +518,7 @@ def _encode_manifest_entry(dest, df, use_short_path):
         src = None
     return json.encode([
         entry_type,
-        dest.strip("/"),
+        dest.rstrip("/"), # do not remove leading /, since we meant(!?) it
```
Comment (JuliaCartwrightKodiak) [inline pkg/private/pkg_files.bzl:521]: are there any of these hacks like this that we can just fix in the vehicle repo?

would ____love____ to drop these changes entirely, or at least move closer to vanilla rules_pkg in some way

Comment (GordonHuangKodiak) [reply pkg/private/pkg_files.bzl:521]: There isn't much that can be done from the vehicle repo. The issue is the unmodified rules_pkg always uses a relative path and will proactively strip leading / from paths. If we don't want to keep this behavior we could remove a lot of the hacks. The side effect is when unpacking tarballs, we'd need to be sure the current directory matches the tarball root directory or else copy the uncompressed files to the correct location.

Comment (JuliaCartwrightKodiak) [reply pkg/private/pkg_files.bzl:521]: have you identified which users in the vehicle repo are relying on this behavior?  that may be a good start.

Comment (GordonHuangKodiak) [reply pkg/private/pkg_files.bzl:521]: I don't know who is using the binary tarballs, but the number of builds that have package_dir = "/something" is quite a lot, 70+.

**Review by AlistairDobkeKodiak — APPROVED** [review]
I tested this on a few archives including the kodiak stack tool.

This tool needed to change its symlink from "./usr/bin/stack" to "/usr/bin/stack" which seems reasonable and more truthful which was the only use of a symlink starting in "./", the other archives with symlinks came out correctly.

**Comment by GordonHuangKodiak** [comment]
Yes, it seems the ./usr/bin/stack is incorrect in the main repo. I have the change as part of PR #20921
https://github.com/KodiakRobotics/vehicle/pull/20921/files#diff-753e9a6362871a3766861740a74f4cedd1957eec57e85375325e0c1b6d87266f

**Pull request merged**